src/vmm/src/devices/virtio/block_common.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+use versionize::{VersionMap, Versionize, VersionizeError, VersionizeResult};
+use versionize_derive::Versionize;
+
+/// Configuration options for disk caching.
+// NOTICE: Any changes to this structure require a snapshot version bump.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Deserialize, Serialize, Versionize)]
+pub enum CacheType {
+    /// Flushing mechanic not will be advertised to the guest driver
```

Comment:
the original messages was
>     /// Flushing mechanic will be advertised to the guest driver, but the operation will be a noop.

is there a reason to change it?

Reply:
The comment was incorrect. If the `CacheType::Unsafe` was used, we did not advertise the `VIRTIO_BLK_F_FLUSH` to the guest.
```rust
        if config.cache_type == CacheType::Writeback {
            avail_features |= 1u64 << VIRTIO_BLK_F_FLUSH;
        }
```
 I will also update the docs, because they are also incorrect.

Reply:
Indeed, https://github.com/firecracker-microvm/firecracker/pull/2702 seems to only update the logic in the code, leaving comments and docs inconsistent. Good finding!

Reply:
Added separate commit to fix the docs